crates/uv-pypi-types/src/conflicting_groups.rs

```diff
@@ -0,0 +1,192 @@
+use uv_normalize::{ExtraName, PackageName};
+
+/// A list of conflicting groups pre-defined by an end user.
+///
+/// This is useful to force the resolver to fork according to extras that have
+/// unavoidable conflicts with each other. (The alternative is that resolution
+/// will fail.)
+#[derive(
+    Debug, Default, Clone, Eq, PartialEq, serde::Deserialize, serde::Serialize, schemars::JsonSchema,
+)]
+pub struct ConflictingGroupList(Vec<ConflictingGroups>);
+
+impl ConflictingGroupList {
+    /// Returns no conflicting groups.
+    ///
+    /// This results in no effect on resolution.
+    pub fn none() -> ConflictingGroupList {
```

Comment:
Nit: I might expect this to be called `empty`.

Reply:
Fine by me! Done.